### PR TITLE
Use scdoc for man pages

### LINKED ISSUE #51: docs: migrate man pages to scdoc
[scdoc](https://git.sr.ht/~sircmpwn/scdoc) is a lightweight man page generator that allows for a much more pleasant writing experience than pure roff. This will make the docs more readable in their plaintext form and allow for easier maintenance.

### PR DESCRIPTION
**Pull request by Ordoviz**
I ran `pandoc -t markdown --atx-headers` on the roff man pages and edited the output manually.

scdoc comes with some downsides, though:
- A new dependency and an extra step between writing and checking the output
- Generated roff files are less readable than hand-written ones
- loss of semantics: In roff we can use `.TP` to create a definition list. In scdoc we need to use indentation.
- scdoc is less established than roff: We cannot use pandoc to convert scdoc files to HTML for example. Text editors and GitHub do not know scdoc: syntax highlighting is non-existing or wrong.

Closes #51

### CODE REVIEW

**Comment by Leon-Plickat** [comment]
> Generated roff files are less readable than hand-written ones

Let's be honest, roff files are basically never readable. The format sucks and is overly complex. Also you can read them with `man`.

Writing is more important than reading.

> loss of semantics: In roff we can use .TP to create a definition list. In scdoc we need to use indentation.

This is a definite improvement, IMO.

>scdoc is less established than roff: We cannot use pandoc to convert scdoc files to HTML for example. Text editors and GitHub do not know scdoc: syntax highlighting is non-existing or wrong.

scdoc is widely available.

Why would we need to convert manpages to html?

Why would you really need syntax highlighting for a simple, markdown-like format? Also: `set ft=markdown` works for me.

File: doc/river.1.scd [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+*river* [*-c* _shell_command_]
+
+# DESCRIPTION
+
+*river* is a dynamic tiling Wayland compositor inspired by dwm and
+bspwm based on wlroots and written in Zig.
+
+# OPTIONS
+
+*-c* _shell_command_
```
Comment (Leon-Plickat) [inline doc/river.1.scd:17]: I think `*-c <shell comand>*` is more idiomatic for scdoc.

File: doc/river.1.scd [hunk 2]
```diff
@@ -0,0 +1,35 @@
+RIVER(1) "github.com/ifreund/river" "General Commands Manual"
```
Comment (Leon-Plickat) [inline doc/river.1.scd:1]: That is not correct scdoc syntax. It should be something like `river(1) ["github.com/ifreund/river" ["General Commands Manual"]]`

File: doc/river.1.scd [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+*-c* _shell_command_
+	Run a shell command or give the path to a script that will be run
+	before river gets started. This is where you define keymaps using
+	*riverctl*(1) and autostart programs. Make sure the script is
+	executable.
+
+# CONFIGURATION
+
+View border width, gap sizes, and programs which should float are
+configured in _Config.zig_. Make your changes and recompile.
```
Comment (Leon-Plickat) [inline doc/river.1.scd:26]: This is no longer true for most configuration options.

File: doc/river.1.scd [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+Experimental XWayland support can be enabled on compile-time with the
+_-Dxwayland=true_ flag.
```
Comment (Leon-Plickat) [inline doc/river.1.scd:29]: Man pages are not compilation instructions. Should be something like "XWayland support may be available if set during compilation."

File: doc/river.1.scd [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+Keyboard layout and other XKB options can be set in _Keyboard.zig_.
```
Comment (Leon-Plickat) [inline doc/river.1.scd:31]: See previous comments.

File: doc/river.1.scd [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+# SEE ALSO
+
+*riverctl*(1), *bspwm*(1), *dwm*(1)
```
Comment (Leon-Plickat) [inline doc/river.1.scd:35]: Do we really want to link to bspwm and dwm? I do not think those man pages help understanding river any better.

File: doc/riverctl.1.scd [hunk 7]
```diff
@@ -0,0 +1,130 @@
+RIVERCTL(1) "github.com/ifreund/river" "General Commands Manual"
```
Comment (Leon-Plickat) [inline doc/riverctl.1.scd:1]: See previous comments.

File: doc/riverctl.1.scd [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+*riverctl* *close*|*declare-mode*|*enter-mode*|*exit*|*focus-output*|
+*focus-view*|*layout*|*map*|*mod-master-count*|*mod-master-factor*|
+*send-to-output*|*set-focused-tags*|*set-view-tags*|*spawn*|*toggle-float*|
+*toggle-focused-tags*|*toggle-view-tags*|*zoom* [_command specific arguments_]
+
+# DESCRIPTION
+
+*riverctl* is a command-line interface inspired by bspc from bspwm
+used to control and configure river.
```
Comment (Leon-Plickat) [inline doc/riverctl.1.scd:16]: Readers of the man page do not really care what the inspiration was. The man page should describe what a program is without falling back to "it's just like $other_program".

**Comment by ifreund** [comment]
Welp, looks like we hit a race condition @Leon-Plickat If you'd like to open a follow up PR with those changes they look good to me.

**Comment by Leon-Plickat** [comment]
Sure, I'll go though them in detail to see what can be improved.

But later, I prefer working on one PR at a time.

File: doc/river.1.scd (hunk 2 above)
Comment (Ordoviz) [reply doc/river.1.scd:1]: The convention of man-pages(7) is to use a capital title, e.g. RIVER not river. The square brackets indicate optional arguments. They are not part of the syntax.

File: doc/river.1.scd (hunk 1 above)
Comment (Ordoviz) [reply doc/river.1.scd:17]: I also see angle brackets used for arguments a lot. Man pages use italics by convention, though.

File: doc/river.1.scd (hunk 3 above)
Comment (Ordoviz) [reply doc/river.1.scd:26]: I will fix the outdated info in a separate PR.

File: doc/river.1.scd (hunk 4 above)
Comment (Ordoviz) [reply doc/river.1.scd:29]: IMO the compile flag is too important to remove it. XWayland support is already stated in the README but mentioning it twice does not hurt. I think we should either keep it or remove the entire sentence.

File: doc/river.1.scd (hunk 5 above)
Comment (Ordoviz) [reply doc/river.1.scd:31]: Agreed, should be replaced with environment variables.

File: doc/riverctl.1.scd (hunk 8 above)
Comment (Ordoviz) [reply doc/riverctl.1.scd:16]: Fair point.

**Pull request merged**